designs/2022-languages/README.md

```diff
@@ -0,0 +1,610 @@
+
+```js
+{
+    "FunctionExpression": {
+        enter(node, parent) {
+            // ...
+        },
+        exit(node, parent) {
+            // ...
+        }
+    }
+}
+```
+
+And if they just wanted to use `:exit`, we could ask them to write this:
+
+```js
+{
+    "FunctionExpression": {
+        exit(node, parent) {
+            // ...
+        }
+    }
+}
+```
+
+This would allow the visitor keys to stay strictly as selectors while still allowing both enter and exit phases of traversal to be hit.
+
+### Does the ESLint team want to create and maintain additional languages?
+
+The next logical step after this refactoring would be to create an additional language to ensure that other languages can be supported. JSON seems like a logical one (as many people seem to think ESLint validates JSON already).
+
+Does the ESLint team want to create an official JSON language plugin? Or any other language plugin? Or do we want to leave additional languages to the community?
+
+I think there is some benefit to having a few officially-supported language plugins that people will likely want, but that also brings additional maintenance burden.
+
+## Help Needed
+
+N/A
+
+## Frequently Asked Questions
+
+### Why allow binary files?
+
+I think there may be a use in validating non-text files such as images, videos, and WebAssembly files. I may be wrong, but by allowing languages to specify that they'd prefer to receive the file in binary, it at least opens up the possibility that people may create plugins that can validate binary files in some way.
+
+### Will the JavaScript language live in the main eslint repo?
+
+As a first step, yes, the JavaScript language object will still live in the main eslint repo. This will allow us to iterate faster and keep an eye on backwards compatibility issues.
+
+In the future, I anticipate moving all JavaScript-specific functionality, including rules, into a separate repository (`eslint/js`) and publishing it as a separate package (`@eslint/js`). That would require some thought around how the website should change and be built as rule documentation would also move into a new repo.
+
+### Why not have `parse()` return a `SourceCode` instead of needing to call another method?
+
+It's important that we are able to determine the amount of time it takes to parse a file in order to optimize performance. Combining the parsing with creating a `SourceCode` object, which may require additional processing of the parse result, means obscuring this important number.
+
+Additionally, `createSourceCode()` allows integrators to do their own parsing and create their own `SourceCode` instances without requiring the use of a specific `parse()` method.
+
+## Related Discussions
+
+* [#6974 Proposal for parser services](https://github.com/eslint/eslint/issues/6974)
```

Comment:
Related and somewhat alternative RFC: https://github.com/eslint/rfcs/pull/56

Reply:
Oh yes, good call!